xdsl/dialects/x86/ops.py

```diff
@@ -86,12 +113,15 @@ def parse(cls, parser: Parser) -> Self:
         pos = parser.pos
         operand_types, result_types = cls.parse_op_type(parser)
         operands = parser.resolve_operands(args, operand_types, pos)
-        return cls.build(
-            operands=operands,
-            result_types=result_types,
-            attributes=attributes,
-            regions=regions,
-        )
+        try:
+            return cls.build(
+                operands=operands,
+                result_types=result_types,
+                attributes=attributes,
+                regions=regions,
+            )
+        except ValueError:
+            raise parser.raise_error(f"{cls}", pos)
```

Comment:
Is this meant to be in this PR?

Reply:
Ah no sorry good catch it was my debugging